FAO Publications.js

```diff
@@ -112,28 +112,45 @@ function scrape(doc, url) {
 		});
 		// url
 		newItem.url = url;
-		// language: according to the last one (old format) or two (new format) letters of PDF file name
-		var langOld = pdfUrl.charAt(pdfUrl.indexOf('pdf') - 2);
-		var langNew = pdfUrl.slice(pdfUrl.indexOf('pdf') - 3, pdfUrl.indexOf('pdf') - 1);
-		if ((langOld == 'a') || (langNew == 'AR') || (langNew == 'ar')) {
+		// language: 2 or 3 letters following ISO 639
+		// indicated by the last 1-3 letters in PDF file name (langCode)
+		// One good example is the various language versions of http://www.fao.org/publications/card/en/c/I2801E
+		var langCode = '';
+		var pdfFilename = pdfUrl.split('/').pop();
+		var pdfFile = pdfFilename.slice(0, pdfFilename.indexOf('.'));
+		// In PDF file name, the immediate char before langCode is always a number.
+		for (let i = pdfFile.length - 1; i >= 0; i--) {
+			if (isNaN(pdfFile.charAt(i))) {
+				langCode = pdfFile.charAt(i) + langCode;
+			}
+			else {
+				break;
+			}
+		}
```

Comment:
This is a bit convoluted and non-idiomatic. You can just use a regexp:

```javascript
var langCode = '';
var matches = pdfUrl.match(/([a-z]+)\.pdf$/i);
if (matches) {
	langCode = matches[1];
}
```

(Could also be restricted further to require a number before the lang code if necessary.)

Reply:
Thanks so much! This is elegant. I have never mastered regexp well so have generally been avoiding it... There's no need to force a number to be exist before lang code for now. I've adjusted the code in the new commit.